patch.mjs

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * Script to create a new patch file.
+ *
+ * Usage: node patch.mjs <patch-name>
+ *
+ * Example: node patch.mjs add-demand-factor-data
+ *
+ *
+ */
+import fs from 'fs';
+
+const date = new Date().toISOString().split('T')[0];
+
+const patchName = process.argv[2];
+
+if (!patchName) {
+  console.error('Patch name is required');
+  process.exit(1);
+}
+
+fs.mkdirSync('patches', { recursive: true });
+fs.writeFileSync(
+  `patches/${date}-${patchName.replace(/ /g, '-').toLowerCase().trim()}.lua`,
```

Comment:
We'll have to introduce any stateful versioning info on the input patch name side here in the event that we have multiple versions of a patch on the same day (not a big deal at all. just making sure we're aligned on that).